src/serialize/serializer.rs

```diff
@@ -577,6 +579,13 @@ impl PyObject {
     {
         let ob_type = ob_type!(self.ptr);
 
+        if py_is!(ob_type, BYTEARRAY_TYPE) {
+            return ByteArray::new(self.ptr).serialize(serializer);
+        }
+        if py_is!(ob_type, MEMORYVIEW_TYPE) {
+            return MemoryView::new(self.ptr).serialize(serializer);
+        }
+
```

Comment:
Can you move the handling of `bytearray` and `memoryview` to the end of both `serialize_unlikely` functions?